Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Gun.h"
+#include "Tile.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    ammunition_ = 1;
+    max_ammunition_ = 1;
+}
+
+bool Gun::UseAmmo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target)
+{
+    id_ptr_on<CubeTile> tile = GetOwner()->GetOwner();
```

Comment:
Oh, you may commit the changes. But can you try to create the separate pull request? (I am just not sure that it is possible). Then I will close this once, because it become too laggy for me cause of a lot of comments.